ci/parse_script.go

```diff
@@ -24,7 +25,18 @@ type AssignmentInfo struct {
 // {{ .AssignmentName }}, {{ .RandomSecret }}, etc.
 func (r RunData) parseScriptTemplate(secret string) (*Job, error) {
 	info := &AssignmentInfo{
-		AssignmentName:     r.Assignment.GetName(),
+		AssignmentName: r.Assignment.GetName(),
+		// TODO(Meling): I have added BranchName here, since we need to checkout the corresponding branch when necessary.
```

Comment:
Please remove this TODO from here and open an issue on the issue tracker to explain what needs to be done. That said, I have some unfinished code locally that should simplify the script, including avoiding this extra if and git checkout.